diff --git a/src/delivery_slot/components/AddNewDeliverySlot.js b/src/delivery_slot/components/AddNewDeliverySlot.js
index e681c4f..429a904 100644
--- a/src/delivery_slot/components/AddNewDeliverySlot.js
+++ b/src/delivery_slot/components/AddNewDeliverySlot.js
@@ -10,8 +10,9 @@ import {
 import { collection, addDoc } from "firebase/firestore";
 import { db } from "../../firebase-config";
 
-function AddNewDeliverySlot() {
-  const [day, setDay] = useState("");
+function AddNewDeliverySlot({data}) {
+  console.log("data",data)
+  const [day, setDay] = useState(data?.day || "" );
   const [timeFrom, setTimeFrom] = useState("");
   const [timeTo, setTimeTo] = useState("");
   const [numDeliveries, setNumDeliveries] = useState("");
diff --git a/src/delivery_slot/components/DeliverySlotList.js b/src/delivery_slot/components/DeliverySlotList.js
index afed98b..a65a8cf 100644
--- a/src/delivery_slot/components/DeliverySlotList.js
+++ b/src/delivery_slot/components/DeliverySlotList.js
@@ -17,7 +17,7 @@ import DeleteIcon from "@mui/icons-material/Delete";
 import OpenInNewIcon from "@mui/icons-material/OpenInNew";
 import { Link } from "react-router-dom";
 
-function DeliverySlotList() {
+function DeliverySlotList({openModal}) {
   const [deliveryslotData, setDeliverySlotData] = useState([]);
   const ref = collection(db, "DeliverySlot");
 
@@ -45,6 +45,7 @@ function DeliverySlotList() {
     await deleteDoc(doc(ref, id));
     setDeliverySlotData(deliveryslotData.filter((row) => row.id !== id));
   };
+  console.log("delivery datasssssssss",deliveryslotData)
 
   return (
     <>
@@ -85,6 +86,7 @@ function DeliverySlotList() {
                         cursor: "pointer",
                       }}
                       className="cursor-pointer"
+                      onClick={()=> openModal(row)}
                     />
                     <DeleteIcon
                       style={{
diff --git a/src/delivery_slot/index.js b/src/delivery_slot/index.js
index c9ee1e6..b00f772 100644
--- a/src/delivery_slot/index.js
+++ b/src/delivery_slot/index.js
@@ -1,5 +1,5 @@
 import React, { useEffect, useState } from "react";
-import { Stack, Typography, Button } from "@mui/material";
+import { Stack, Typography, Button, Box, Modal } from "@mui/material";
 import TextField from "@mui/material/TextField";
 import AddNewDeliverySlot from "../delivery_slot/components/AddNewDeliverySlot";
 import DeliverySlotList from "../delivery_slot/components/DeliverySlotList";
@@ -17,6 +17,7 @@ function DeliverySlots() {
     const [addNewDeliverySlot, setAddNewDeliverySlot] = useState(false);
     const handleOpen = () => setAddNewDeliverySlot(true);
     const handleClose = () => setAddNewDeliverySlot(false);
+    const [deliveryModalData, setDeliveryModalData] = useState(null)
     const [options, setOptions] = useState([]);
     const rows = useAppStore((state) => state.rows);
     const setRows = useAppStore((state) => state.setRows);
@@ -43,6 +44,14 @@ function DeliverySlots() {
         }
     };
 
+    const modal = () => (
+        <Modal onClose={() => setAddNewDeliverySlot(false)} open={addNewDeliverySlot}>
+          <Box sx={{ width: '50%', margin: '0 auto', top: '50%' }}>
+            <AddNewDeliverySlot closeModal={()=>setAddNewDeliverySlot(false)} data={deliveryModalData} />
+          </Box>
+        </Modal>
+      );
+
 
     const actionBar = () => (
         <>
@@ -75,33 +84,11 @@ function DeliverySlots() {
     );
     return (
         <>
-            <PageTemplate actionBar={actionBar()} title={"Delivery Slots"}>
-                <DeliverySlotList />
+            <PageTemplate modal={modal()} actionBar={actionBar()} title={"Delivery Slots"}>
+                <DeliverySlotList openModal={(row)=> {setDeliveryModalData(row)
+                handleOpen()}}/>
             </PageTemplate>
-            {addNewDeliverySlot && (
-                <div
-                    style={{
-                        position: "absolute",
-                        top: "50%",
-                        left: "50%",
-                        transform: "translate(-50%, -50%)",
-                        backgroundColor: "#fff",
-                        padding: "20px",
-                        boxShadow: "0px 5px 15px rgba(0, 0, 0, 0.1)",
-                        borderRadius: "5px",
-                        border: "1px solid #000",
-                        zIndex: "999",
-                        overflowY: "scroll",
-                        maxHeight: "80vh", // set a maximum height to prevent the pop-up from taking up the entire screen
-                    }}
-                >
-                    <div style={{ display: "flex", justifyContent: "flex-end" }}>
-                        <button onClick={handleClose}>X</button>
-                    </div>
-                    <h2 style={{ textAlign: "center" }}>Add Delivery Slot</h2>
-                    <AddNewDeliverySlot />
-                </div>
-            )}
+
         </>
     );
 }
